vshard/storage/init.lua

```diff
@@ -1486,7 +1557,14 @@ local function bucket_force_create_impl(first_bucket_id, count)
     box.begin()
     local limit = consts.BUCKET_CHUNK_SIZE
     for i = first_bucket_id, first_bucket_id + count - 1 do
-        _bucket:insert({i, consts.BUCKET.ACTIVE})
+        -- Buckets are protected with on_replace trigger, which
+        -- prohibits the creation of the ACTIVE buckets from nowhere,
+        -- as this is done only for bootstrap and not during the normal
+        -- work of vshard. So, as we don't want to disable the protection
+        -- of the buckets for the whole replicaset for bootstrap, a bucket's
+        -- status must go the following way: none -> RECEIVING -> ACTIVE.
+        _bucket:insert({i, consts.BUCKET.RECEIVING})
+        _bucket:replace({i, consts.BUCKET.ACTIVE})
```

Comment:
I remember my biggest concern in my own patch was about `bucket_force_create()` and `bucket_force_drop()`. Currently both of them can actually fail. So they are not much of a "force". The creator is good enough, but `bucket_force_drop()` becomes useless. It won't be able to delete anything except `GARBAGE` which are deleted automatically anyway.

However I don't mind if we just won't do anything with it for now. Doing it properly is too complicated. Not only you would have to gracefully walk the state machine from any state to `GARBAGE`, but also take care of the bucket refs, which can't be done transactionally.

Reply:
I suppose we should make a new issue related to the proper `bucket_force_drop` and tag it as feature. Currently `bucket_force_create` is crucial, as it's used during cluster bootstrap, speaking of `bucket_force_drop`, it can be implemented later

Reply:
I don't mind.